### PR TITLE
Dionisia - Edges - js-adagrams

### PR DESCRIPTION
**Pull request by larachan15**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | The logic seemed to be pretty consistent. I'm still struggling with the syntax however.
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | No, I used the same methods to find information. Stack Overflow was helpful. I found the Math.random method which I used to shuffle the letters.
What was a challenge you were able to overcome on this assignment? | I realize I still struggle with hashes and how to assign and access info in a hash. Trying to do it in a new language, even though it's a similar technique made it clear that I need to review this further. Also writing functions in JavaScript is still a little confusing to me.
What is your favorite thing about learning a new programming language? | It was nice to use existing knowledge of Ruby to understand how the basic syntax works. JavaScript is not as intuitive as Ruby is for me though.
What is your least favorite thing about learning a new programming language? | I made a lot of syntax errors and I don't know how to read through the stack trace very well. Not having good debugging tools yet was also frustrating.
Do you have any recommendations on how we could improve this project for the next cohort? | This intro to JavaScript felt pretty rushed. It might have been nice to have started the Friday prior to let some of this information sit. Or maybe we could have pushed the JQuery intro to next week so that we could get more of the fundamentals of JS down first.

### CODE REVIEW

**Comment by tildeee** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | x
Small commits with meaningful commit messages | x, it would be better to have a commit message that talked more about what the code changes were and not "wave 2" etc
**Code Requirements** |
`drawLetters` method | x
Uses appropriate data structure to store the letter distribution | x
All tests for `drawLetters` pass | x
`usesAvailableLetters` method | x
All tests for `usesAvailableLetters` pass | x
`scoreWord` method | x
Uses appropriate data structure to store the letter scores | x
All tests for `scoreWord` pass | x
`highestScoreFrom` method | x
Appropriately handles edge cases for tie-breaking logic | x
All tests for `highestScoreFrom` pass | x
**Overall** |

Dionisia, wonderful job on this project! I really like your code style; the approach/solution you did for each problem is a nice, elegant, short one. Nicely done!

I only have a few comments on this that are very minor and optional. Overall, great work

File: src/adagrams.js [hunk 1]
```diff
@@ -1,7 +1,119 @@
...
+      if (lettersInHandMap[letter]) {
+        lettersInHandMap[letter] += 1;
+      }
+      else {
+        lettersInHandMap[letter] = 1;
+      }
+    });
+
+    for (let i = 0; i < input.length; i++) {
+      if (lettersInHandMap[input[i]]) {
```
Comment (tildeee) [inline src/adagrams.js:52]: nicely done! This relies on the fact that JavaScript says that 0 is falsey and any other integer is truthy

File: src/adagrams.js [hunk 2]
```diff
@@ -1,7 +1,119 @@
...
+      "F": 4, "H": 4, "V": 4, "W": 4, "Y": 4,
+      "K": 5,
+      "J": 8, "X": 8,
+      "Q": 10, "Z": 10
+    };
+    let score = 0;
+    // let upcaseLetter = letter.toUpperCase();
+
+    for (let i = 0; i < word.length; i++) {
+      let upcaseLetter = word.toUpperCase();
```
Comment (tildeee) [inline src/adagrams.js:77]: does `upcaseLetter` describe what `word.toUpperCase()` means? Does this need to be reassigned each iteration of the for loop? Could this be refactored into a different kind of loop like a forEach loop?
